config/config_file.go

```diff
@@ -65,7 +65,7 @@ func (l configFileLoader) Configure(cfg *Config) error {
 	// Skip loading config file if some authentication is already explicitly
 	// configured directly in the config by a user.
 	// See: https://github.com/databricks/databricks-sdk-go/issues/304
-	if cfg.Profile == "" && (l.isAnyAuthConfigured(cfg) || cfg.IsAzure()) {
+	if cfg.Profile == "" && (l.isAnyAuthConfigured(cfg) || cfg.IsAws() || cfg.IsAzure() || cfg.IsGcp()) {
```

Comment:
but do we need to list all clouds here? It's equivalent of using any cloud

Reply:
I'm also that it will solve this https://github.com/databricks/terraform-provider-databricks/issues/2149 - as we're using non-cloud URL in tests

Reply:
Good point -- made the field checks explicit.